### PR TITLE
Document optional dependency CMake build options

### PR DESCRIPTION
**Pull request by jiridanek**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -311,4 +311,22 @@ Older versions of `CMake`` only honor this option with the Intel compiler on Lin
 This option *breaks* safe pointers, resulting in crashes, therefore is suitable only for debugging.
 When combined with `-DRUNTIME_CHECK=asan`, the pointer breakages are much less frequent.
 
+|`-DBUILD_TESTING=OFF`
+|Excludes project's tests from the build.
```
Comment (jiridanek) [inline README.adoc:315]: This is a standard CMake feature, see https://cmake.org/cmake/help/latest/module/CTest.html.

Dispatch does not appear to fully support it, though it should. Reported as #172.

File: README.adoc [hunk 2]
```diff
@@ -311,4 +311,22 @@ Older versions of `CMake`` only honor this option with the Intel compiler on Lin
...
+
+|`-DBUILD_BENCHMARKS=ON`
+|Benchmarking tests will be built.
+The `libbenchmark` library is required by the benchmarks.
+
+|`-DUSE_JEMALLOC=ON`
+|The jemalloc library will be linked, replacing the standard C library memory allocation functions with
+its own optimized implementations.
```
Comment (jiridanek) [inline README.adoc:323]: This will be added shortly, #166.

File: README.adoc (hunk 1 above)
Comment (jiridanek) [reply README.adoc:315]: To be fixed by #177

File: README.adoc (hunk 2 above)
Comment (jiridanek) [reply README.adoc:323]: To be merged in #176

File: README.adoc (hunk 1 above)
Comment (jiridanek) [reply README.adoc:315]: Merged

File: README.adoc (hunk 2 above)
Comment (jiridanek) [reply README.adoc]: Removed from PR

**Review by kgiusti — APPROVED** [review]

**Pull request merged**